lib/signs/utilities/messages.ex

```diff
@@ -258,6 +287,34 @@ defmodule Signs.Utilities.Messages do
     end
   end
 
+  defp overnight_period?(
+         current_time,
+         first_scheduled_departure,
+         last_scheduled_departure,
+         recent_departures,
+         service_ended
+       ) do
+    last_actual_departure =
+      if service_ended do
+        recent_departures
+        |> Enum.max_by(fn {_key, datetime} -> datetime end, fn -> {nil, nil} end)
+        |> elem(1)
```

Comment:
Hmm so this does give use the "last actual departure" relative to a point in time, but I don't think it tells us whether or not that most recent departure was flagged as a last trip. It's possible that even after the last scheduled trip there are multiple trains that run before the "actual last trip" finally passes through a station.

Reply:
Hmm valid point. So if service has ended, that means that **a** last trip has departed from the station recently. It _could_ be possible that an extraneous trip passes through that is not necessarily marked as a last trip (I think likely due to human error if anything), but I think your point stands that if service is considered ended, then using the most recent departure should be fine!